script.service.kodi.callbacks/default.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+#     Copyright (C) 2015 KenV99
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+debug = False  # TODO: check
+testdebug = False  # TODO: check
+testTasks = False  # TODO: check
+branch = 'master'
+
+from resources.lib.utils.debugger import startdebugger
+
+if debug:
+    startdebugger()
+
+import os
+import sys
+import threading
+import xbmc
+import xbmcaddon
+import xbmcgui
+import resources.lib.pubsub as PubSub_Threaded
+from resources.lib.kodilogging import KodiLogger
+from resources.lib.publisherfactory import PublisherFactory
+from resources.lib.subscriberfactory import SubscriberFactory
+from resources.lib.settings import Settings
+from resources.lib.utils.kodipathtools import translatepath
+from resources.lib.utils.poutil import KodiPo
+
+kodipo = KodiPo()
+_ = kodipo.getLocalizedString
+log = KodiLogger.log
+
+try:
+    __version__ = xbmcaddon.Addon().getAddonInfo('version')
+except RuntimeError:
+    try:
+        __version__ = xbmcaddon.Addon('script.service.kodi.callbacks').getAddonInfo('version')
+    except RuntimeError:
+        __version__ = 'ERROR getting version'
+
+
+def createUserTasks():
+    paths = [translatepath('special://addondata')]
+    paths.append(os.path.join(paths[0], 'lib'))
+    paths.append(os.path.join(paths[1], 'usertasks'))
+    for path in paths:
+        if not os.path.isdir(path):
+            try:
+                os.mkdir(path)
```

Comment:
Ok. Is that required? I can do that but then that requires a lot of mocking when running tests outside of Kodi.